src/include/wiredtiger.in

```diff
@@ -3682,6 +3704,502 @@ struct __wt_extractor {
 	int (*terminate)(WT_EXTRACTOR *extractor, WT_SESSION *session);
 };
 
+#if !defined(SWIG)
+/*! WT_FILE_SYSTEM::open_file type: checkpoint */
+#define	WT_FILE_TYPE_CHECKPOINT	1
+/*! WT_FILE_SYSTEM::open_file type: read/write data object */
+#define	WT_FILE_TYPE_DATA	2
+/*! WT_FILE_SYSTEM::open_file type: directory */
+#define	WT_FILE_TYPE_DIRECTORY	3
+/*! WT_FILE_SYSTEM::open_file type: log file */
+#define	WT_FILE_TYPE_LOG	4
+/*! WT_FILE_SYSTEM::open_file type: regular file */
+#define	WT_FILE_TYPE_REGULAR	5
+
+/*! WT_FILE_SYSTEM::open_file flags: create if does not exist */
+#define	WT_OPEN_CREATE		0x001
+/*! WT_FILE_SYSTEM::open_file flags: direct I/O requested */
+#define	WT_OPEN_DIRECTIO	0x002
+/*! WT_FILE_SYSTEM::open_file flags: error if exclusive use not available */
+#define	WT_OPEN_EXCLUSIVE	0x004
+#ifndef DOXYGEN
+#define	WT_OPEN_FIXED		0x008	/* Path not home relative (internal) */
+#endif
+/*! WT_FILE_SYSTEM::open_file flags: open is read-only */
+#define	WT_OPEN_READONLY	0x010
+
+/*!
+ * The interface implemented by applications to provide a custom file system
+ * implementation.
+ *
+ * Applications register implementations with WiredTiger by calling
+ * WT_CONNECTION::add_file_system.  See @ref custom_file_systems for more
+ * information.
+ *
+ * @snippet ex_file_system.c WT_FILE_SYSTEM register
+ */
+struct __wt_file_system {
+	/*!
+	 * Return the list of file names for the named directory.
+	 *
+	 * This method is not required by applications that do not configure
+	 * logging, and should be set to NULL when not required by the
+	 * application.
```

Comment:
As discussed, I think we should just require this.  There's no way for a filesystem implementer to know whether an application will configure logging.